### PR TITLE
fix(sdk): fix chain metadata shape when adding new chain for user

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David <david@umaproject.org>

**Motivation**
web3 rpc is very strict when sending data to user to add a new chain, causing errors when including unexpected props. 

**Summary**

added utility to strip unneeded props from object before requesting user add the new chain

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [x]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/sdk/src/oracle/utils.ts [hunk 1]
```diff
@@ -189,3 +191,10 @@ export class TransactionConfirmer {
     return false;
   }
 }
+
+export function chainConfigToChainMetadata(config: ChainConfig): ChainMetadata {
+  // linters hate this, but we just need to strip out these vars for valid chain metadata
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (mrice32) [inline packages/sdk/src/oracle/utils.ts:197]: You could also add this option to the eslint config so this doesn't cause issues here or in the future!
https://eslint.org/docs/rules/no-unused-vars#ignorerestsiblings

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/sdk/src/oracle/utils.ts (hunk 1 above)
Comment (daywiss) [reply packages/sdk/src/oracle/utils.ts:197]: i generally do like this rule!

Comment (daywiss) [reply packages/sdk/src/oracle/utils.ts:197]: oh thanks for correcting me, this rule is specificallly for this case, i will add that

**Pull request merged**